src/odemis/acq/stitching/_tiledacq.py

```diff
@@ -77,11 +83,14 @@ def __init__(self, streams, stage, area, overlap, settings_obs=None, log_path=No
         :param future: (ProgressiveFuture or None) future to track progress, pass None for estimation only
         :param zlevels: (list(float) or None) focus z positions required zstack acquisition.
            Currently, can only be used if focusing_method == MAX_INTENSITY_PROJECTION.
+           If focus_points is defined, zlevels is adjusted relative to the focus_points.
         :param registrar: (REGISTER_*) type of registration method
         :param weaver: (WEAVER_*) type of weaving method
         :param focusing_method: (FocusingMethod) Defines when will the autofocuser be run.
            The autofocuser uses the first stream with a .focuser.
            If MAX_INTENSITY_PROJECTION is used, zlevels must be provided too.
+        :param focus_points: (list of tuples) list of focus points corresponding to the known (x, y, z) at good focus.
+              If None, the focus will not be adjusted based on the stage position.
         """
         self._future = future
```

Comment:
self._focus_points is defined below in your code but in if condition. In test cases, it fails as it is expecting self._focus_points.
```suggestion
        self._focus_points = focus_points
        self._future = future
```